src/Features/LanguageServer/Protocol/Handler/SpellCheck/AbstractSpellCheckingHandler.cs

```diff
@@ -139,9 +140,22 @@ protected AbstractSpellCheckHandler()
             {
                 if (requestParams.TextDocument != null)
                 {
-                    var document = context.Solution.GetDocument(requestParams.TextDocument);
+                    var document = await context.Solution.GetDocumentAsync(requestParams.TextDocument, cancellationToken).ConfigureAwait(false);
+
                     if (document != null)
+                    {
                         result[document] = requestParams;
+                    }
+                    else
+                    {
+                        context.TraceInformation($"Clearing spans for removed document: {requestParams.TextDocument.Uri}");
+
+                        // Client is asking server about a document that no longer exists (i.e. was removed/deleted from
+                        // the workspace). Report a (null-spans, null-result-id) response to the client as that means
+                        // they should just consider the file deleted and should remove all spans information they've
+                        // cached for it.
+                        progress.Report(CreateReport(requestParams.TextDocument, ranges: null, resultId: null));
+                    }
```

Comment:
was this a goof on my part?

Reply:
question.  why merge these?  is removing all the old, then adding the new/changed not a nice way to ahndle things?

Reply:
General cleanup: both functions looped to the same list (although in slightly different ways), both called GetDocument, and then one only operated if the document was null, one if it wasn't. It didn't even cross my mind this might have been an intentional pattern -- it looked like a refactoring that ended up in a silly place and nobody noticed.